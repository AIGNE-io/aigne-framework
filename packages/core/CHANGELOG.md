## 1.1.0-beta.17 (2025-3-18)

- chore: add support for esm module


## [1.2.0](https://github.com/AIGNE-io/aigne-framework/compare/core-next-v1.1.0...core-next-v1.2.0) (2025-03-18)


### Features

* add Agent FunctionAgent AIAgent MCPAgent and ExecutionEngine ([4d2a5a1](https://github.com/AIGNE-io/aigne-framework/commit/4d2a5a1b3366b8f935f50a0937c2da6e49073348))
* add OrchestratorAgent in agent library ([25a5e9e](https://github.com/AIGNE-io/aigne-framework/commit/25a5e9e6c60d747c8bf484ac884b31dc02c14757))
* add sequential and parallel helper function ([a295697](https://github.com/AIGNE-io/aigne-framework/commit/a295697b5694754e02954fc5c7f382a3b219a3ab))
* add support for MCP resources ([1ded2fb](https://github.com/AIGNE-io/aigne-framework/commit/1ded2fbf222fa8984e75df0852ff384524f73b04))
* **prompt-builder:** support chat history in PromptBuilder ([6ca05f2](https://github.com/AIGNE-io/aigne-framework/commit/6ca05f28eddb683a4f1e228865f8bbf8a8e190f1))
* support run puppeteer example chat loop in terminal ([85ce7f8](https://github.com/AIGNE-io/aigne-framework/commit/85ce7f8de8b443c86e50815dd7bcab99f869c4ce))
* use PromptBuilder instead of string instructions ([e4cb2cb](https://github.com/AIGNE-io/aigne-framework/commit/e4cb2cb6baf4f9bcef390567a4a174e9246c29a3))


### Bug Fixes

* **AIAgent:** should pass both arguments (model generated) and input (user input) to tool ([c49d64e](https://github.com/AIGNE-io/aigne-framework/commit/c49d64ee35f7efd83b0f82f43205bb1c40f999e8))
* **MCP:** catch list resource error treat as empty list ([1885fab](https://github.com/AIGNE-io/aigne-framework/commit/1885fab3585e0dd1467b127e5b47cd0b98282bab))
* use text resource from MCP correctly ([8b9eba8](https://github.com/AIGNE-io/aigne-framework/commit/8b9eba83352ec096a2a5d4f410d4c4bde7420bce))

## 1.1.0-beta.16 (2025-3-18)

- chore: add puppeteer in linux need docker_container

## 1.1.0-beta.15 (2025-3-18)

- chore: make coverage report as text to terminal
- chore: update contributing docs

## 1.1.0-beta.14 (2025-3-18)

- chore(example): add code-execution example

## 1.1.0-beta.13 (2025-3-18)

- feat: add OrchestratorAgent in agent library

## 1.1.0-beta.12 (2025-3-14)

- chore(example): add concurrency reflection handoff workflow examples

## 1.1.0-beta.11 (2025-3-14)

- feat(core): add sequential and parallel helper function
- chore(examples): add workflow-sequential example

## 1.1.0-beta.10 (2025-3-13)

- chore: ensure required environment variables have values

## 1.1.0-beta.9 (2025-3-13)

- fix(MCP): catch list resource error treat as empty list

## 1.1.0-beta.8 (2025-3-13)

- fix(AIAgent): should pass both arguments (model generated) and input (user input) to tool
- chore(examples): add workflow-router example
- chore(examples): rename examples puppeteer-mcp-server and sqlite-mcp-server to mcp-server-puppeteer and mcp-server-sqlite

## 1.1.0-beta.7 (2025-3-13)

- chore: rename @aigne/core to @aigne/core-next

## 1.1.0-beta.6 (2025-3-13)

- chore(examples): default enable mcp debug message for examples

## 1.1.0-beta.5 (2025-3-13)

- feat: support chat history in PromptBuilder
- feat: add `prompts` for MCPAgent to consume prompts from MCP server
- chore: add sqlite-mcp-server example
- test: add more unit test cases

## 1.1.0-beta.4 (2025-3-12)

- feat: support run puppeteer example chat loop in terminal

## 1.1.0-beta.3 (2025-3-11)

- chore: set module type for core package

## 1.1.0-beta.2 (2025-3-11)

- feat: use PromptBuilder instead of string instructions
- refactor: use tools instead of skills
- chore(examples): add puppeteer-mcp-server example

## 1.1.0-beta.1 (2025-3-11)

- feat: add Agent FunctionAgent AIAgent MCPAgent and ExecutionEngine
