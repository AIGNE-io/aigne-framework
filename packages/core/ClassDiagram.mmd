classDiagram
    class PromptBuilderBuildOptions {
        +Context context
        +Agent agent
        +object input
        +ChatModel model
    }

    class Prompt {
        +List~object~ messages
        +List~Agent~ tools
        +object toolChoice
        +object responseFormat
    }

    PromptBuilder ..> PromptBuilderBuildOptions: dependency
    PromptBuilder ..> Prompt: dependency
    class PromptBuilder {
        +build(PromptBuilderBuildOptions options) Prompt
    }

    class Model {
        +call(Prompt input) object*
    }

    ChatModel --|> Model: inheritance
    class ChatModel {
    }

    ImageModel --|> Model: inheritance
    class ImageModel {
    }

    Agent --|> EventEmitter: inheritance
    class Agent {
        +string name
        +string description
        +object inputSchema
        +object outputSchema
        +List~string~ inputTopic
        +List~string~ nextTopic
        +List~Agent~ tools
        +Map~string, Agent~ skills

        +call(string input, Context context) object
        +call(object input, Context context) object

        +process(object input, Context context) object*
        -verifyInput() void
        -verifyOutput() void
    }

    AIAgent --|> Agent: inheritance
    AIAgent *.. PromptBuilder: composition
    class AIAgent {
        +ChatModel model
        +string instructions
        +string outputKey
        +object toolChoice
        +PromptBuilder promptBuilder
    }

    ImageAgent --|> Agent: inheritance
    ImageAgent *.. PromptBuilder: composition
    class ImageAgent {
        +ImageModel model
        +string instructions
        +PromptBuilder promptBuilder
    }

    FunctionAgent --|> Agent: inheritance
    class FunctionAgent {
        +Function fn
    }

    RPCAgent --|> Agent: inheritance
    class RPCAgent {
        +string url
    }

    MCPAgent --|> Agent: inheritance
    MCPAgent *.. MCPClient: composition
    class MCPAgent {
        +MCPClient client
    }

    class MCPClient {
    }

    class Message {
        +object output
    }

    MessageQueue ..> Message: dependency
    class MessageQueue {
        +publish(string topic, Message message) void
        +subscribe(string topic, Function callback) void
        +unsubscribe(string topic, Function callback) void
    }

    class History {
        +string id
        +string agentId
        +object input
        +object output
    }

    Context *.. ChatModel: composition
    Context *.. ImageModel: composition
    Context --|> MessageQueue: inheritance
    Context ..> History: dependency
    class Context {
        +ChatModel model
        +ImageModel imageModel
        +List~Agent~ tools

        +getHistories(string agentId, int limit) List~History~
        +addHistory(History history) void
        +publish(string topic, Message message) void
        +subscribe(string topic, Function callback) void
        +unsubscribe(string topic, Function callback) void
    }

    class EventEmitter {
        +on(): void
        +emit(): void
    }

    UserAgent --|> Agent: inheritance
    class UserAgent {
    }

    class ExecutionEngineRunOptions {
        +boolean concurrency
    }

    ExecutionEngine --|> Context: inheritance
    ExecutionEngine --|> EventEmitter: inheritance
    ExecutionEngine ..> UserAgent: dependency
    ExecutionEngine ..> ExecutionEngineRunOptions: dependency
    class ExecutionEngine {
        +run(Agent agent) UserAgent
        +run(string input) object
        +run(object input) object
        +run(object input, Agent ...agents) object
        +run(object input, ExecutionEngineRunOptions options, Agent ...agents) object
    }
