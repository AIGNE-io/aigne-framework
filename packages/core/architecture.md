## Class Definitions

```mermaid
classDiagram
    class PromptBuilderBuildOptions {
        +Context context
        +Agent agent
        +object input
        +ChatModel model
    }

    class Prompt {
        +List~object~ messages
        +List~Agent~ tools
        +object toolChoice
        +object responseFormat
    }

    PromptBuilder ..> PromptBuilderBuildOptions: dependency
    PromptBuilder ..> Prompt: dependency
    class PromptBuilder {
        +build(PromptBuilderBuildOptions options) Prompt
    }

    class Model {
        +call(Prompt input) object*
    }

    ChatModel --|> Model: inheritance
    class ChatModel {
    }

    ImageModel --|> Model: inheritance
    class ImageModel {
    }

    Agent --|> EventEmitter: inheritance
    class Agent {
        +string name
        +string description
        +object inputSchema
        +object outputSchema
        +List~string~ inputTopic
        +List~string~ nextTopic
        +List~Agent~ tools

        +call(string input, Context context) object
        +call(object input, Context context) object

        +process(object input, Context context) object*
        -verifyInput() void
        -verifyOutput() void
    }

    AIAgent --|> Agent: inheritance
    AIAgent *.. PromptBuilder: composition
    class AIAgent {
        +ChatModel model
        +string instructions
        +string outputKey
        +object toolChoice
        +PromptBuilder promptBuilder
    }

    ImageAgent --|> Agent: inheritance
    ImageAgent *.. PromptBuilder: composition
    class ImageAgent {
        +ImageModel model
        +string instructions
        +PromptBuilder promptBuilder
    }

    FunctionAgent --|> Agent: inheritance
    class FunctionAgent {
        +Function fn
    }

    RPCAgent --|> Agent: inheritance
    class RPCAgent {
        +string url
    }

    MCPAgent --|> Agent: inheritance
    MCPAgent *.. MCPClient: composition
    class MCPAgent {
        +MCPClient client
    }

    class MCPClient {
    }

    class Message {
        +object output
    }

    MessageQueue ..> Message: dependency
    class MessageQueue {
        +publish(string topic, Message message) void
        +subscribe(string topic, Function callback) void
        +unsubscribe(string topic, Function callback) void
    }

    class History {
        +string id
        +string agentId
        +object input
        +object output
    }

    Context *.. ChatModel: composition
    Context *.. ImageModel: composition
    Context --|> MessageQueue: inheritance
    Context ..> History: dependency
    class Context {
        +ChatModel model
        +ImageModel imageModel
        +List~Agent~ tools

        +getHistories(string agentId, int limit) List~History~
        +addHistory(History history) void
        +publish(string topic, Message message) void
        +subscribe(string topic, Function callback) void
        +unsubscribe(string topic, Function callback) void
    }

    class EventEmitter {
        +on(): void
        +emit(): void
    }

    UserAgent --|> Agent: inheritance
    class UserAgent {
    }


    AIGNE --|> Context: inheritance
    AIGNE --|> EventEmitter: inheritance
    AIGNE ..> UserAgent: dependency
    class AIGNE {
        +call(Agent agent) UserAgent
        +call(Agent agent, string input) object
        +call(Agent agent, object input) object
    }
```
