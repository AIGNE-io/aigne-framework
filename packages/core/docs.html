<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport"
    content="width=device-width, initial-scale=1, minimum-scale=1.0, shrink-to-fit=no, viewport-fit=cover">
  <meta name="description" content="">
  <title></title>

  <link rel="stylesheet" href="//cdn.jsdelivr.net/npm/docsify/themes/buble.css" />

  <!-- Custom Styles -->
  <style>
    :root {
      /* --theme-hue: 325; */
    }
  </style>
</head>

<body>
  <div id="app"></div>

  <script>
    // Docsify Configuration
    window.$docsify = {
      name: 'AIGNE Framework',
      loadSidebar: true,
      alias: {
        '/.*/_sidebar.md': '/_sidebar.md'
      },
      relativePath: true
    };
  </script>

  <!-- Required -->
  <script src="https://cdn.jsdelivr.net/npm/docsify@4/lib/docsify.min.js"></script>

  <!-- Recommended -->
  <script src="https://cdn.jsdelivr.net/npm/docsify@4/lib/plugins/search.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/docsify@4/lib/plugins/zoom-image.min.js"></script>

  <script src="//cdn.jsdelivr.net/npm/prismjs@1/components/prism-typescript.min.js"></script>
</body>

</html>
