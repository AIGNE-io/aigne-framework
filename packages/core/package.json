{
  "name": "@aigne/core",
  "version": "1.11.0",
  "description": "AIGNE core library for building AI-powered applications",
  "publishConfig": {
    "access": "public"
  },
  "author": "Arcblock <blocklet@arcblock.io> https://github.com/blocklet",
  "homepage": "https://github.com/AIGNE-io/aigne-framework",
  "license": "Elastic-2.0",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/AIGNE-io/aigne-framework"
  },
  "files": [
    "lib/cjs",
    "lib/dts",
    "lib/esm",
    "LICENSE",
    "README.md",
    "CHANGELOG.md"
  ],
  "type": "module",
  "main": "./lib/cjs/index.js",
  "module": "./lib/esm/index.js",
  "types": "./lib/dts/index.d.ts",
  "exports": {
    ".": {
      "import": "./lib/esm/index.js",
      "require": "./lib/cjs/index.js",
      "types": "./lib/dts/index.d.ts"
    },
    "./*": {
      "import": "./lib/esm/*",
      "require": "./lib/cjs/*",
      "types": "./lib/dts/*"
    }
  },
  "typesVersions": {
    "*": {
      ".": [
        "./lib/dts/index.d.ts"
      ],
      "agents/*": [
        "./lib/dts/agents/*"
      ],
      "execution-engine/*": [
        "./lib/dts/execution-engine/*"
      ],
      "loader/*": [
        "./lib/dts/loader/*"
      ],
      "models/*": [
        "./lib/dts/models/*"
      ],
      "prompt/*": [
        "./lib/dts/prompt/*"
      ],
      "utils/*": [
        "./lib/dts/utils/*"
      ]
    }
  },
  "scripts": {
    "lint": "tsc --noEmit",
    "build": "tsc --build scripts/tsconfig.build.json",
    "clean": "rimraf lib test/coverage",
    "prepublishOnly": "run-s clean build",
    "test": "bun --cwd test test",
    "test:coverage": "bun --cwd test test --coverage --coverage-reporter=lcov --coverage-reporter=text",
    "postbuild": "echo '{\"type\": \"module\"}' > lib/esm/package.json && echo '{\"type\": \"commonjs\"}' > lib/cjs/package.json"
  },
  "dependencies": {
    "@aigne/json-schema-to-zod": "^1.3.3",
    "@modelcontextprotocol/sdk": "^1.9.0",
    "@types/debug": "^4.1.12",
    "camelize-ts": "^3.0.0",
    "debug": "^4.4.0",
    "fast-deep-equal": "^3.1.3",
    "inquirer": "^12.5.2",
    "mustache": "^4.2.0",
    "nanoid": "^5.1.5",
    "p-retry": "^6.2.1",
    "uuid": "^11.1.0",
    "yaml": "^2.7.1",
    "zod": "^3.24.2",
    "zod-to-json-schema": "^3.24.5"
  },
  "peerDependencies": {
    "@anthropic-ai/sdk": "^0.39.0",
    "@google/generative-ai": "^0.24.0",
    "openai": "^4.87.3"
  },
  "devDependencies": {
    "@anthropic-ai/sdk": "^0.39.0",
    "@google/generative-ai": "^0.24.0",
    "@types/bun": "^1.2.9",
    "@types/express": "^5.0.1",
    "@types/mustache": "^4.2.5",
    "@types/node": "^22.14.1",
    "detect-port": "^2.1.0",
    "express": "^5.1.0",
    "npm-run-all": "^4.1.5",
    "openai": "^4.94.0",
    "rimraf": "^6.0.1",
    "typescript": "^5.8.3"
  }
}
