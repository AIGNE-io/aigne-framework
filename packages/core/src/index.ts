export * from "./agent";
export * from "./constants";
export * from "./context";
export * from "./definitions/data-type";
export * from "./definitions/data-type-schema";
export * from "./definitions/memory";
export * from "./definitions/open-api";
export * from "./function-agent";
export * from "./function-runner";
export * from "./llm-agent";
export * from "./llm-decision-agent";
export * from "./llm-model";
export * from "./llm-models/gemini-llm-model";
export * from "./llm-models/openai-llm-model";
export * from "./local-function-agent";
export * from "./memorable";
export * from "./open-api-agent";
export * from "./pipeline-agent";
export * from "./runnable";
export * from "./runtime";
export * from "./utils";
