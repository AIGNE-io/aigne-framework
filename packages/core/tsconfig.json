{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "paths": {
      "@aigne/core": ["./src"],
      "@aigne/core/*": ["./src/*"]
    }
  },
  "exclude": ["node_modules", "lib"],
  "include": ["src", "test"]
}
