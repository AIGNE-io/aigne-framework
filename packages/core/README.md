# @aigne/core

[![GitHub star chart](https://img.shields.io/github/stars/AIGNE-io/aigne-framework?style=flat-square)](https://star-history.com/#AIGNE-io/aigne-framework)
[![Open Issues](https://img.shields.io/github/issues-raw/AIGNE-io/aigne-framework?style=flat-square)](https://github.com/AIGNE-io/aigne-framework/issues)
[![codecov](https://codecov.io/gh/AIGNE-io/aigne-framework/graph/badge.svg?token=DO07834RQL)](https://codecov.io/gh/AIGNE-io/aigne-framework)
[![NPM Version](https://img.shields.io/npm/v/@aigne/core)](https://www.npmjs.com/package/@aigne/core)
[![Elastic-2.0 licensed](https://img.shields.io/npm/l/@aigne/core)](https://github.com/AIGNE-io/aigne-framework/blob/main/LICENSE)

Core library of [AIGNE Framework](https://github.com/AIGNE-io/aigne-framework) for building AI-powered applications.

## Introduction

`@aigne/core` is the foundation component of [AIGNE Framework](https://github.com/AIGNE-io/aigne-framework), providing the essential modules and tools needed to build AI-driven applications. This package implements the core functionalities of the framework, including agent systems, aigne environment, model integrations, and workflow pattern support.

## Features

* **Multiple AI Model Support**: Built-in support for OpenAI, Gemini, Claude, Nova, and other mainstream AI models, easily extensible to support additional models
* **Agent System**: Powerful agent abstractions supporting AI agents, function agents, MCP agents, and more
* **AIGNE Environment**: Flexible handling communication between agents and workflow execution
* **Workflow Patterns**: Support for sequential, concurrent, routing, handoff, and other workflow patterns
* **MCP Protocol Integration**: Seamless integration with external systems through the Model Context Protocol
* **TypeScript Support**: Comprehensive type definitions providing an excellent development experience

## Installation

The AIGNE Framework depends on the core package `@aigne/core` and model packages (such as `@aigne/openai`). You can choose the appropriate installation command based on your package manager. After installation, you can start building your own Agents.

### Using npm

```bash
npm install @aigne/core

# model packages
npm install @aigne/openai
```

### Using yarn

```bash
yarn add @aigne/core

# model packages
yarn add @aigne/openai
```

### Using pnpm

```bash
pnpm add @aigne/core

# model packages
pnpm add @aigne/openai
```

### available models

```typescript
import { AnthropicChatModel } from "@aigne/anthropic";
import { BedrockChatModel } from "@aigne/bedrock";
import { DeepSeekChatModel } from "@aigne/deepseek";
import { GeminiChatModel } from "@aigne/gemini";
import { OllamaChatModel } from "@aigne/ollama";
import { OpenRouterChatModel } from "@aigne/open-router";
import { OpenAIChatModel } from "@aigne/openai";
import { XAIChatModel } from "@aigne/xai";
```

## Basic Usage

```typescript
import { AIAgent, AIGNE } from "@aigne/core";
import { OpenAIChatModel } from "@aigne/openai";

// Create AI model instance
const model = new OpenAIChatModel({
  apiKey: process.env.OPENAI_API_KEY,
  model: process.env.DEFAULT_CHAT_MODEL || "gpt-4o-mini",
});

// Create AI agent
  const agent = AIAgent.from({
    instructions: "You are a helpful assistant",
    inputKey: "message",
    outputKey: "text",
  });

// AIGNE: Main execution engine of AIGNE Framework.
const aigne = new AIGNE({ model });

// Use the AIGNE to invoke the agent
const response = await aigne.invoke(, agent,
  "Hello, can you help me write a short article?",
);
console.log(response);
// { text: "xxxxx"}
```

## License

Elastic-2.0
