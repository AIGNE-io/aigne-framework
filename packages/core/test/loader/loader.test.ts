import { expect, spyOn, test } from "bun:test";
import assert from "node:assert";
import { join } from "node:path";
import { ChatModel, createMessage } from "@aigne/core";
import { load } from "@aigne/core/loader/index.js";
import { nanoid } from "nanoid";

test("loader should load agents correctly", async () => {
  const engine = await load({ path: join(import.meta.dirname, "./test-agent-library/chat") });

  expect(engine.agents.length).toBe(1);

  const chat = engine.agents[0];
  expect(chat).toEqual(
    expect.objectContaining({
      name: "chat",
    }),
  );
  assert(chat, "chat agent should be defined");

  expect(chat.tools.length).toBe(1);
  expect(chat.tools[0]).toEqual(expect.objectContaining({ name: "calculate" }));

  expect(engine.model).toBeInstanceOf(ChatModel);
  assert(engine.model, "model should be defined");

  spyOn(engine.model, "call")
    .mockReturnValueOnce(
      Promise.resolve({
        toolCalls: [
          {
            id: nanoid(),
            type: "function",
            function: { name: "calculate", arguments: { a: 1, b: 2 } },
          },
        ],
      }),
    )
    .mockReturnValueOnce(Promise.resolve({ text: "1 + 2 = 3" }));

  const result = await engine.call(chat, "1 + 2 = ?");
  expect(result).toEqual(expect.objectContaining(createMessage("1 + 2 = 3")));
});

test("loader should error if agent file is not supported", async () => {
  const engine = load({
    path: join(import.meta.dirname, "./test-agent-library/invalid-agent-file"),
  });

  expect(engine).rejects.toThrow("Unsupported agent file type");
});
