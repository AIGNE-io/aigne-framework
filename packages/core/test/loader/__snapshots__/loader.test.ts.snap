// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`AIGNE.load should load agents correctly 1`] = `
{
  "agents": [
    {
      "description": "Chat agent",
      "name": "chat",
      "taskTitle": "process: {{message}}",
    },
    {
      "description": "Chat agent",
      "name": "chat-with-prompt",
      "taskTitle": undefined,
    },
    {
      "description": "Test team agent",
      "name": "test-team-agent",
      "taskTitle": undefined,
    },
    {
      "description": undefined,
      "name": "test-image-agent",
      "taskTitle": undefined,
    },
    {
      "description": undefined,
      "name": "test-relative-prompt-paths",
      "taskTitle": undefined,
    },
  ],
  "cli": {
    "agents": [
      {
        "agent": {
          "description": "Chat agent",
          "name": "chat",
        },
        "commands": undefined,
      },
      {
        "agent": {
          "description": undefined,
          "name": "b",
        },
        "commands": undefined,
      },
      {
        "agent": {
          "description": undefined,
          "name": "a",
        },
        "commands": [
          {
            "agent": {
              "description": undefined,
              "name": "a1",
            },
            "commands": [
              {
                "agent": {
                  "description": undefined,
                  "name": "a11",
                },
                "commands": undefined,
                "url": "test-cli-agents/a-1-1.yaml",
              },
              {
                "agent": {
                  "description": undefined,
                  "name": "a12",
                },
                "commands": undefined,
                "description": "A agent from a-1-2.yaml",
                "name": "a12-renamed",
                "url": "test-cli-agents/a-1-2.yaml",
              },
            ],
            "url": "test-cli-agents/a-1.yaml",
          },
          {
            "agent": {
              "description": undefined,
              "name": "a2",
            },
            "commands": undefined,
          },
        ],
        "description": "A agent from a.yaml",
        "name": "a-renamed",
        "url": "test-cli-agents/a.yaml",
      },
    ],
  },
  "mcpServer": {
    "agents": [
      {
        "description": "Chat agent",
        "name": "chat",
      },
    ],
  },
  "skills": [
    {
      "description": "This agent evaluates JavaScript code.",
      "name": "evaluateJs",
    },
  ],
}
`;
